actors/migration/nv3/miner.go

```diff
@@ -1,27 +1,426 @@
+			return nil
+		})
+	})
+
+	claims, err := adt.AsMap(store, powSt.Claims)
+	if err != nil {
+		return err
+	}
+
+	var existing power.Claim
+	found, err := claims.Get(abi.AddrKey(a), &existing)
+	if err != nil {
+		return err
+	}
+	if found {
+		activePower = activePower.Sub(miner.NewPowerPair(existing.RawBytePower, existing.QualityAdjPower))
+	}
+	if err := powSt.AddToClaim(store, a, activePower.Raw, activePower.QA); err != nil {
+		return err
+	}
+
+	powSt.Claims, err = claims.Root()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Updates the expiration queue by correcting any duplicate entries and their fallout.
+// If no changes need to be made cid.Undef will be returned.
+func (m *minerMigrator) correctExpirationQueue(exq miner.ExpirationQueue, sectors miner.Sectors,
+	allTerminated bitfield.BitField, allFaults bitfield.BitField, sectorSize abi.SectorSize,
+) (cid.Cid, miner.PowerPair, miner.PowerPair, bitfield.BitField, error) {
+	// processed expired sectors includes all terminated and all sectors seen in earlier expiration sets
+	processedExpiredSectors := allTerminated
+	terminatedAsNonFaults := bitfield.New()
+
+	alteredExpirationSets := make(map[abi.ChainEpoch]miner.ExpirationSet)
+	erasedFaults := bitfield.New()
+	var exs miner.ExpirationSet
+
+	// check for faults that need to be erased
+	err := exq.ForEach(&exs, func(epoch int64) error {
+		// handle faulty sector expirations that have already been processed
+		earlyDuplicates, err := bitfield.IntersectBitField(exs.EarlySectors, processedExpiredSectors)
+		if err != nil {
+			return err
+		} else if empty, err := earlyDuplicates.IsEmpty(); err != nil {
+			return err
+		} else if !empty {
+			alteredExpirationSets[abi.ChainEpoch(epoch)] = exs
+			exs.EarlySectors, err = bitfield.SubtractBitField(exs.EarlySectors, earlyDuplicates)
+			if err != nil {
+				return err
+			}
+
+			terminatedActive, err := bitfield.IntersectBitField(earlyDuplicates, allTerminated)
+			if err != nil {
+				return err
+			}
+			terminatedAsNonFaults, err = bitfield.MergeBitFields(terminatedAsNonFaults, terminatedActive)
+			if err != nil {
+				return err
+			}
+		}
+
+		onTimeDuplicates, err := bitfield.IntersectBitField(exs.OnTimeSectors, processedExpiredSectors)
+		if err != nil {
+			return err
+		} else if empty, err := onTimeDuplicates.IsEmpty(); err != nil {
+			return err
+		} else if !empty {
+			alteredExpirationSets[abi.ChainEpoch(epoch)] = exs
+			exs.OnTimeSectors, err = bitfield.SubtractBitField(exs.OnTimeSectors, onTimeDuplicates)
+			if err != nil {
+				return err
+			}
+
+			terminatedActive, err := bitfield.IntersectBitField(onTimeDuplicates, allTerminated)
+			if err != nil {
+				return err
+			}
+			terminatedAsNonFaults, err = bitfield.MergeBitFields(terminatedAsNonFaults, terminatedActive)
+			if err != nil {
+				return err
+			}
+		}
+
+		if erasedFaults, err = bitfield.MultiMerge(erasedFaults, earlyDuplicates, onTimeDuplicates); err != nil {
```

Comment:
I don't think you know whether the power was in the faulty sum or not or not, but luckily this value appears unused and can be removed.